File: backend/turip-app/src/main/java/turip/article/controller/dto/response/ArticleResponse.java
package turip.article.controller.dto.response;

import java.time.LocalDateTime;
import java.util.List;
import turip.article.domain.Article;
import turip.place.controller.dto.response.PlaceResponse;

public record ArticleResponse(
        Long id,
        String title,
        String subtitle,
        String content,
        String thumbnailUrl,
        List<String> tags,
        AuthorResponse author,
        List<PlaceResponse> places,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    public static ArticleResponse of(
            Article article,
            String thumbnailUrl,
            List<String> tagNames,
            List<PlaceResponse> places
    ) {
        return new ArticleResponse(
                article.getId(),
                article.getTitle(),
                article.getSubtitle(),
                article.getContent(),
                thumbnailUrl,
                tagNames,
                AuthorResponse.from(article.getAuthor()),
                places,
                article.getCreatedAt(),
                article.getUpdatedAt()
        );
    }
}

File: mobile/core/data/src/commonMain/kotlin/com/on/turip/core/data/dto/article/ArticleResponse.kt
package com.on.turip.core.data.dto.article

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ArticleResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("title")
    val title: String,
    @SerialName("subtitle")
    val subtitle: String,
    @SerialName("thumbnailUrl")
    val thumbnailUrl: String,
    @SerialName("tags")
    val tags: List<String>,
    @SerialName("author")
    val author: ArticleAuthorResponse?,
    @SerialName("createdAt")
    val createdAt: String,
)
